packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Password from 'react-jsonschema-form/lib/components/widgets/PasswordWidget';
+import classNames from 'classnames';
+import { renderActions } from '../../Form';
+import theme from './TogglePasswordWidget.scss';
+
+export default class TogglePasswordWidget extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onToggle = this.onToggle.bind(this);
+		this.state = {
+			type: props.type?props.type:'password',
+		};
+	}
+
+	onToggle() {
+		this.setState({ type: this.state.type === 'text' ? 'password' : 'text' });
+	}
+
+	handleActionClick(onClick) { // eslint-disable-line class-methods-use-this
+		if (onClick) {
+			return () => onClick();
+		}
+		return () => {
+		};
```

Comment:
this looks weird.
handle doesn't not handle the click it returns a function.
getOnClick ?

Reply:
same code as Form.js (handleActionClick(onClick))